pkg/k8splan/watcher.go

```diff
@@ -0,0 +1,196 @@
+package k8splan
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/rancher/lasso/pkg/cache"
+	"github.com/rancher/lasso/pkg/client"
+	"github.com/rancher/lasso/pkg/controller"
+	"github.com/rancher/system-agent/pkg/applyinator"
+	"github.com/rancher/system-agent/pkg/config"
+	"github.com/rancher/system-agent/pkg/prober"
+	corecontrollers "github.com/rancher/wrangler/pkg/generated/controllers/core/v1"
+	"github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+const (
+	appliedChecksumKey   = "applied-checksum"
+	appliedOutputKey     = "applied-output"
+	probeStatusesKey     = "probe-statuses"
+	probePeriodKey       = "probe-period"
+	planKey              = "plan"
+	enqueueAfterDuration = "5s"
+)
+
+func Watch(ctx context.Context, applyinator applyinator.Applyinator, connInfo config.ConnectionInfo) error {
```

Comment:
Does this need to return an error since it can't?

Reply:
error removed